homeassistant/components/sun/sensor.py

```diff
@@ -26,19 +26,14 @@
 ENTITY_ID_SENSOR_FORMAT = SENSOR_DOMAIN + ".sun_{}"
 
 
-@dataclass
-class SunEntityDescriptionMixin:
-    """Mixin for required Sun base description keys."""
+@dataclass(kw_only=True)
```

Comment:
https://developers.home-assistant.io/blog/2023/12/11/entity-description-changes/

Reply:
Can't set `frozen=True`, mypy is complaining (Cannot inherit frozen dataclass from a non-frozen one), as #105512 is missing

Reply:
I think, we should then wait for #105512